## --------------------------------------------------------------------------
## 예외발생
## - 문법, 실행 시 아무런 문제 없음
## - 개발자의 필요에 의해서 발생시키는 예외
## - 개발자가 예외 객체 생성
## - 형식/문법 : raise Exception("예외 메세지")
## --------------------------------------------------------------------------
## 나눗셈에서 발생하는 오류 : 0으로 나눔
## - 데이터 : 리스트 [10, 20, 30]
## - 나눌값 : 입력 받기
## --------------------------------------------------------------------------

## --------------------------------------------------------------------------
## 함수기능 : 3의 배수 입력 및 검사
## 함수이름 : three_multiple
## 매개변수 : 없음
## 결과반환 : 없음
## --------------------------------------------------------------------------
def three_multiple():
    num = int(input("3의 배수 입력: "))
    if num%3 != 0 :
        raise Exception("3의 배수 아님")
    print(f'{num} 은 3의 배수')
    

## --------------------------------------------------------------------------
## 함수 호출
## --------------------------------------------------------------------------
     
try :
    three_multiple()        
except Exception as e :
    print("EROOR 발생 : " , e)

else :
    pass
    


